flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -249,4 +250,22 @@ private void seizeCapacity() {
         setCurrentContext(storedContext);
         inFlightRecordNum.incrementAndGet();
     }
+
+    /**
+     * A helper to request a {@link StateRequestType#SYNC_POINT} and run a callback if it finishes.
```

Comment:
"if it finishes" is a little bit ambiguous, how about "if the record is not blocked"?